### PR TITLE
Fixes #34630 - Errata Filters Date do not show "Start Date" & "End Date"

### PR DESCRIPTION
**Pull request by Andrewgdewar**
#### What are the changes introduced in this pull request?

- Updated invalid date error text
Screen Shot 2022-03-16 at 2 07 07 PM

- Made the placeholder text dynamic based on focus to act as label.
Screen Shot 2022-03-16 at 2 07 36 PM

#### What are the testing steps for this pull request?

- Create both types of errata filters
- Enter an invalid date, view copy.
- onFocus the placeholder text should convert to the expected date format
- onBlur the placeholder should return to its label (start or end date) if input left blank

### CODE REVIEW

**Comment by jeremylenz** [comment]
date_filled-out

After the dates are entered, it's no longer apparent what the fields are for. This means one cannot "confirm the current user inputs" as mentioned in the original bug.

**Comment by MariSvirik** [comment]
@Andrewgdewar @jeremylenz 
What about this :

Screenshot 2022-03-18 at 13 38 13

Screenshot 2022-03-18 at 13 41 32


• Add "to" between the fields
• Add "from" to the dropdown to display Issued from/Updated from ( it's always going to be range in this page) OR instead add "From" before the start date picker.
• Let's use from/to in a chip as well: "Updated from MM/DD/YYYY to MM/DD/YYYY"

Helper text if entering invalid format: "Enter a valid format: MM/DD/YYYY" 
Fix sentence case for Start date/End date

**Comment by jeremylenz** [comment]
> What about this

It both allows us to benefit from the work @Andrewgdewar has already put in here, and improves on the design to address the original concern. I think this will be a great solution! :+1:

**Comment by Andrewgdewar** [comment]
> @Andrewgdewar @jeremylenz What about this :
> 
> Screenshot 2022-03-18 at 13 38 13
> 
> Screenshot 2022-03-18 at 13 41 32
> 
> • Add "to" between the fields • Add "from" to the dropdown to display Issued from/Updated from ( it's always going to be range in this page) OR instead add "From" before the start date picker. • Let's use from/to in a chip as well: "Updated from MM/DD/YYYY to MM/DD/YYYY"
> 
> Helper text if entering invalid format: "Enter a valid format: MM/DD/YYYY" Fix sentence case for Start date/End date

Sounds good, with the exception that one does not enter a format, they enter text. A format is MM/DD/YYYY.  Alternative copy @jeremylenz?

**Comment by jeremylenz** [comment]
> Sounds good, with the exception of the helper text change: one does not enter a format, they enter text. A format is MM/DD/YYYY

How about `Enter a date in the format MM/DD/YYYY` ?

**Comment by jeremylenz** [comment]
> How about `Enter a date in the format MM/DD/YYYY` ?

Not to complicate things, but what if the user's locale is set to one that uses a different date format? Will this date picker always use the same format? Will it use the local format and this label will be inaccurate?

**Comment by Andrewgdewar** [comment]
\

> > How about `Enter a date in the format MM/DD/YYYY` ?
> 
> Not to complicate things, but what if the user's locale is set to one that uses a different date format? Will this date picker always use the same format? Will it use the local format and this label will be inaccurate?

Currently all of the pickers in the app MM/DD/YYYY (Without translation)

**Comment by jeremylenz** [comment]
> Currently all of the pickers in the app MM/DD/YYYY (Without translation)

too bad! I suppose that's not in scope here though, so I won't block this on it.

**Comment by MariSvirik** [comment]
> > How about `Enter a date in the format MM/DD/YYYY` ?
> 
> Not to complicate things, but what if the user's locale is set to one that uses a different date format? Will this date picker always use the same format? Will it use the local format and this label will be inaccurate?

I thought it would change automatically based on the user preferences. My bad. In any case, the accepted/valid format inside the date picker and the suggested correct format displayed in the helper text should be the same. Consistency FTW!

The point of the helper text was to give users precise information about what went wrong and how to fix it.
This is the copy from the [PF4 guidelines](https://www.patternfly.org/v4/components/date-picker/design-guidelines) - "Enter a valid date: MM/DD/YYYY" But I'm fine with @jeremylenz's suggestion.

**Comment by Andrewgdewar** [comment]
Screen Shot 2022-03-18 at 11 53 51 AM
Screen Shot 2022-03-18 at 11 26 05 AM
Screen Shot 2022-03-18 at 11 52 11 AM

File: webpack/scenes/ContentViews/Details/Filters/CVErrataDateFilterContent.js [hunk 1]
```diff
@@ -179,49 +184,51 @@ const CVErrataDateFilterContent = ({
...
+                >
                   <DatePicker
                     aria-label="start_date_input"
                     value={startDate}
+                    invalidFormatText={invalidDateFormat}
                     dateFormat={dateFormat}
                     onChange={setStartDate}
                     dateParse={dateParse}
-                    placeholder="MM/DD/YYYY"
+                    placeholder={startEntry ? 'MM/DD/YYYY' : __('Start Date')}
```
Comment (jeremylenz) [inline webpack/scenes/ContentViews/Details/Filters/CVErrataDateFilterContent.js:203]:
```suggestion
                    placeholder={startEntry ? 'MM/DD/YYYY' : __('Start date')}
```

File: webpack/scenes/ContentViews/Details/Filters/CVErrataDateFilterContent.js [hunk 2]
```diff
@@ -179,49 +184,51 @@ const CVErrataDateFilterContent = ({
...
+                >
                   <DatePicker
                     aria-label="end_date_input"
                     value={endDate}
+                    invalidFormatText={invalidDateFormat}
                     dateFormat={dateFormat}
                     onChange={setEndDate}
                     dateParse={dateParse}
-                    placeholder="MM/DD/YYYY"
+                    placeholder={endEntry ? 'MM/DD/YYYY' : __('End Date')}
```
Comment (jeremylenz) [inline webpack/scenes/ContentViews/Details/Filters/CVErrataDateFilterContent.js:225]:
```suggestion
                    placeholder={endEntry ? 'MM/DD/YYYY' : __('End date')}
```

File: webpack/scenes/ContentViews/Details/Filters/CVErrataDateFilterContent.js [hunk 3]
```diff
@@ -103,6 +106,7 @@ const CVErrataDateFilterContent = ({
   }, [showAffectedRepos, repositories.length]);
 
   const tabTitle = inclusion ? __('Included errata') : __('Excluded errata');
+  const invalidDateFormat = __('Enter a valid date: MM/DD/YYYY');
```
Comment (jeremylenz) [inline webpack/scenes/ContentViews/Details/Filters/CVErrataDateFilterContent.js:109]: :+1: :+1:

File: webpack/scenes/ContentViews/Details/Filters/CVErrataIDFilterContent.js [hunk 4]
```diff
@@ -349,54 +353,67 @@ const CVErrataIDFilterContent = ({
...
+                    >
                       <DatePicker
                         aria-label="start_date_input"
+                        invalidFormatText={invalidDateFormat}
                         value={startDate}
                         dateFormat={dateFormat}
                         onChange={setValidStartDate}
                         dateParse={dateParse}
-                        placeholder="MM/DD/YYYY"
+                        placeholder={startEntry ? 'MM/DD/YYYY' : __('Start Date')}
```
Comment (jeremylenz) [inline webpack/scenes/ContentViews/Details/Filters/CVErrataIDFilterContent.js:372]:
```suggestion
                        placeholder={startEntry ? 'MM/DD/YYYY' : __('Start date')}
```

File: webpack/scenes/ContentViews/Details/Filters/CVErrataIDFilterContent.js [hunk 5]
```diff
@@ -349,54 +353,67 @@ const CVErrataIDFilterContent = ({
...
+                    >
                       <DatePicker
                         aria-label="end_date_input"
                         value={endDate}
+                        invalidFormatText={invalidDateFormat}
                         dateFormat={dateFormat}
                         onChange={setValidEndDate}
                         dateParse={dateParse}
-                        placeholder="MM/DD/YYYY"
+                        placeholder={endEntry ? 'MM/DD/YYYY' : __('End Date')}
```
Comment (jeremylenz) [inline webpack/scenes/ContentViews/Details/Filters/CVErrataIDFilterContent.js:393]:
```suggestion
                        placeholder={endEntry ? 'MM/DD/YYYY' : __('End date')}
```

**Review by jeremylenz — CHANGES_REQUESTED** [review]

**Review by jeremylenz — APPROVED** [review]
Changes look good, thanks @Andrewgdewar !

ACK :+1:

**Comment by Andrewgdewar** [comment]
> 
@jeremylenz 
Thank you good sir!

**Pull request merged**